Source/WebCore/rendering/svg/RenderSVGResourceMarker.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) Research In Motion Limited 2009-2010. All rights reserved.
+ * Copyright (C) 2022, 2023, 2024 Igalia S.L.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * along with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#pragma once
+
+#include "RenderSVGResourceContainer.h"
+#include "SVGMarkerTypes.h"
+
+namespace WebCore {
+
+class GraphicsContext;
+class SVGMarkerElement;
+
+class RenderSVGResourceMarker final : public RenderSVGResourceContainer {
+    WTF_MAKE_ISO_ALLOCATED(RenderSVGResourceMarker);
+public:
+    RenderSVGResourceMarker(SVGMarkerElement&, RenderStyle&&);
+    virtual ~RenderSVGResourceMarker();
+
+    inline SVGMarkerElement& markerElement() const;
```

Comment:
Does not seem to be used outside of RenderSVGResourceMarker, so could be private.